### ISSUE TITLE
[Bug]: When turning off "Show ice icon" and the turning it back on, the icon doesn't automatically repopulate

### ISSUE THREAD

**Issue by philocalyst** [jordanbaird/Ice#329]
### Check Existing Issues

- [X] I have checked existing issues, and this issue is not a duplicate

### Description

When toggling, it takes a reselect of the item or hover for the icon the reappear. 

### Steps to Reproduce

1. General
2. Show ice icon (toggle)
3. Show ice icon (toggle)
4. Gasp

### Ice Version

10.5

### macOS Version

14.6.1 (23G93)

**Comment by jordanbaird** [comment]
Interesting, I can't reproduce this. What icon image do you have selected?

**Comment by philocalyst** [comment]
Tried again, and starting to think it's more related to slowdowns with the apps/routines than a logic issue. Maybe I was running some intense tasks that stole processing power or I was on low battery mode? I can still get it to happen (inconsistently) if I toggle fast enough. I've also generally had performance issues with the app, so it may be indicative of a deeper problem, either with my computer or the app.

Let me know if you need any more information!

=====

**Comment by jordanbaird** [comment]
@philocalyst How old is your computer? Ice uses SwiftUI, which tends to be kind of performance intensive on macOS, unfortunately. There are probably some optimizations that can be done, but for the vast majority of users, this shouldn't be a problem.

**Comment by philocalyst** [comment]
I run a MBPro 14’ M3 Pro, so I feel confident saying the stutters aren’t a computer issue. It is unfortunate how difficult swift UI makes it to make applications run smoothly, I wish I knew anything about it to contribute. 

Appreciate your work tons! Glad it’s a subtle issue

▰▰▰▰▰
Miles Wirht 🙃

**Comment by jordanbaird** [comment]
@philocalyst Yeah, that definitely doesn't sound like a problem with your machine. I'm thinking that it's probably an upstream issue on Apple's end, but I'll keep this open for now and look into it further when I get the chance.
